plugin/environment.go

```diff
@@ -177,39 +178,41 @@ func (env *Environment) Activate(id string) (reterr error) {
 	if pluginInfo.Manifest.Webapp != nil {
 		bundlePath := filepath.Clean(pluginInfo.Manifest.Webapp.BundlePath)
 		if bundlePath == "" || bundlePath[0] == '.' {
-			return fmt.Errorf("invalid webapp bundle path")
+			return nil, false, fmt.Errorf("invalid webapp bundle path")
 		}
 		bundlePath = filepath.Join(env.pluginDir, id, bundlePath)
 
 		webappBundle, err := ioutil.ReadFile(bundlePath)
 		if err != nil {
-			return errors.Wrapf(err, "unable to read webapp bundle: %v", id)
+			return nil, false, errors.Wrapf(err, "unable to read webapp bundle: %v", id)
 		}
 
-		err = ioutil.WriteFile(fmt.Sprintf("%s/%s_bundle.js", env.webappPluginDir, id), webappBundle, 0644)
+		pluginInfo.Manifest.Webapp.BundleHash = md5.Sum(webappBundle)
```

Comment:
Why md5?

Reply:
It's just an odd pick. We don't need crypto hashes here. Why pick a broken crypto hash? In this situation we have used FNV before: https://golang.org/pkg/hash/fnv/

Reply:
MD5 is cryptographically broken, yes, but there's no security threat at play here. MD5 is typically fast (hardware support), familiar (gut instinct when squinting at hexadecimal characters is to assume MD5 or SHA1) and implemented in lots of places (`md5sum` on the command line, etc.).

I'm not sure that FNC is an improvement over MD5 in this context. I also couldn't find us using `fnv.New128` anywhere for file hashing -- just `fnv.New32a` for seeding user colours and hash mapping hubs by user, both of which appear to require constrained ranges for the hash output.

I'm 0/5 on what we actually use, since it's a trivial code change either way, but am happy to be convinced otherwise :)

Reply:
Using the benchmark test:
```golang
package main                                                                                                                                                                        
                                                                                                                                                                                      
  import (                                                                                                                                                                            
      "crypto/md5"                                                                                                                                                                    
      "crypto/sha256"                                                                                                                                                                 
      "hash/fnv"                                                                                                                                                                      
      "io/ioutil"                                                                                                                                                                     
      "testing"                                                                                                                                                                       
  )                                                                                                                                                                                   
                                                                                                                                                                                      
  var input []byte                                                                                                                                                                    
                                                                                                                                                                                      
  func init() {                                                                                                                                                                       
      data, err := ioutil.ReadFile("testbundle.js")                                                                                                                                   
      if err != nil {                                                                                                                                                                 
          panic(err)                                                                                                                                                                  
      }                                                                                                                                                                               
      input = data                                                                                                                                                                    
  }                                                                                                                                                                                   
                                                                                                                                                                                      
  func BenchmarkMd5(b *testing.B) {                                                                                                                                                   
      for n := 0; n < b.N; n++ {                                                                                                                                                      
          md5.Sum(input)                                                                                                                                                              
      }                                                                                                                                                                               
  }                                                                                                                                                                                   
                                                                                                                                                                                      
  func BenchmarkSHA256(b *testing.B) {                                                                                                                                                
      for n := 0; n < b.N; n++ {                                                                                                                                                      
          sha256.Sum256(input)                                                                                                                                                        
      }                                                                                                                                                                               
  }                                                                                                                                                                                   
                                                                                                                                                                                      
  func BenchmarkFNV64A(b *testing.B) {                                                                                                                                                
      hash := fnv.New64a()                                                                                                                                                            
      for n := 0; n < b.N; n++ {                                                                                                                                                      
          hash.Sum(input)                                                                                                                                                             
      }                                                                                                                                                                               
  }                                                                                                                                                                                   
                                                                                                                                                                                      
  func BenchmarkFNV128A(b *testing.B) {                                                                                                                                               
      hash := fnv.New128a()                                                                                                                                                           
      for n := 0; n < b.N; n++ {                                                                                                                                                      
          hash.Sum(input)                                                                                                                                                             
      }                                                                                                                                                                               
  }                                                                                                                                                                                   
```

On my machine running an i7-7700K I get:
```
goos: linux
goarch: amd64
BenchmarkMd5-8       	    5000	    380557 ns/op
BenchmarkSHA256-8    	    2000	    684145 ns/op
BenchmarkFNV64A-8    	1000000000	         2.67 ns/op
BenchmarkFNV128A-8   	1000000000	         2.99 ns/op
PASS
ok  	9.618s
```

Really there was no competition. Crypto hashes are designed to be slow.

Reply:
Fair enough :)